components/xiaomi_ylkg07yl/__init__.py

```diff
@@ -136,7 +136,12 @@ async def to_code(config):
             sens = await sensor.new_sensor(conf)
             cg.add(getattr(var, f"set_{key}_sensor")(sens))
 
-    for action in ON_PRESS_ACTIONS:
+    if CONF_ON_PRESS in config:
```

Comment:
I would like to avoid the code duplication here by keeping the old structure. What do you think about it?